peepingtom/datablocks/datablock.py

```diff
@@ -18,9 +18,29 @@ def __init__(self, name=None, volume=None, peeper=None, parent=None):
         self._name = name
         self.depictors = []
         self.alchemists = []
-        self.volume = volume
+        self._volume = volume
         self.peeper = peeper
 
+    @property
+    def parent(self):
+        """
+        points to the parent in case of a view, or to self otherwise
+        """
+        return self._parent or self
```

Comment:
I don't understand how this works, could you explain?

Reply:
Oh! You might not know that in python `or` does not simply return True or False, but returns the first element that evaluated to True. This means that:
```
print(0 or 'Thing')
```
will actually print `Thing`, not True. This means that the above always returns the first ancestor no matter what, because self._parent is set to None (which evaluates to False) by default, and therefore when a Datablock has no parent, it returns `self` instead.